### ISSUE TITLE
How to disable IOS notification permission on launch?

### ISSUE THREAD

**Issue by sowsan** [MaikuB/flutter_local_notifications#1314]
In IOS notification is getting whenever launch the app. Is there a way to disable it and handle it on demand within the app? There are reported issues on the same and all are reported related to flutter_local_notifications. Thanks for your help!

https://stackoverflow.com/questions/66694451/flutter-block-push-notification-permission-on-ios-at-startup
https://github.com/FirebaseExtended/flutterfire/issues/5897

**Comment by MaikuB** [comment]
Yes, the answer was in the stack overflow link you've posted, specifically this answer https://stackoverflow.com/a/66705721. This is also referenced in the readme [here](https://github.com/MaikuB/flutter_local_notifications/tree/master/flutter_local_notifications#ios-all-supported-versions-and-macos-1014-requesting-notification-permissions). Have you tried following those steps?

**Comment by toklive** [comment]
Great thank you so much. We will test it out

**Comment by desmeit** [comment]
I try to call the permissions at a later point with:

```
void requestIOSPermissions() {
    _notifications
        .resolvePlatformSpecificImplementation<
            IOSFlutterLocalNotificationsPlugin>()
        ?.requestPermissions(
          alert: true,
          badge: true,
          sound: true,
        );
  }
```

Is it possible to get a callback if the user clicked the alert?

**Comment by MaikuB** [comment]
`requestPermissions` returns a boolean to indicate if the permissions have been granted or not
